platform/qt/src/qmapboxgl.cpp

```diff
@@ -785,6 +894,21 @@ void QMapboxGL::setLayoutProperty(const QString& layer_, const QString& property
     }
 }
 
+/*!
+    Sets a paint \a property \a value to an existing \a layer_. The \a property string can be any
+    as defined by \l {https://www.mapbox.com/mapbox-gl-style-spec/} {the vector tiles specification}
+    for paint properties.
+
+    The argument \a klass_ is deprecated and is used for defining the style class for the paint
```

Comment:
Does the compiler used with Qt support a deprecation attribute? We could remove the `klass_` parameter from this method and overload the method with a deprecated one that takes `klass_`.